cylc/flow/config.py

```diff
@@ -729,20 +729,22 @@ def process_initial_cycle_point(self):
             WorkflowConfigError - if it fails to validate
         """
         orig_icp = self.cfg['scheduling']['initial cycle point']
-        if orig_icp is None:
-            if self.cfg['scheduling']['cycling mode'] == INTEGER_CYCLING_TYPE:
-                orig_icp = '1'
+        if self.cycling_type == INTEGER_CYCLING_TYPE:
+            if orig_icp is None:
+                icp = orig_icp = '1'
             else:
+                icp = orig_icp
```

Comment:
Or also

```py
            if orig_icp is None:
                orig_icp = '1'
            icp = orig_icp
```

Either way is OK for me :man_shrugging:

Reply:
Your way is better :grin: